tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.cs

```diff
@@ -138,32 +138,37 @@ internal readonly partial struct SecurityCoordinator
         return null;
     }
 
-    private static Dictionary<string, object> ExtractHeaders(ICollection<string> keys, Func<string, object> getHeaderValue)
+    private static Dictionary<string, object>? ExtractHeaders(ICollection<string> keys, Func<string, object> getHeaderValue)
     {
-        var headersDic = new Dictionary<string, object>(keys.Count);
-        foreach (var key in keys)
+        if (keys.Count > 0)
         {
-            var currentKey = key ?? string.Empty;
-            if (!currentKey.Equals("cookie", StringComparison.OrdinalIgnoreCase))
+            var headersDic = new Dictionary<string, object>(keys.Count);
+            foreach (var key in keys)
             {
-                currentKey = currentKey.ToLowerInvariant();
-                var value = getHeaderValue(currentKey);
-
-                if (value is not null)
+                var currentKey = key ?? string.Empty;
+                if (!currentKey.Equals("cookie", StringComparison.OrdinalIgnoreCase))
                 {
-                    if (!headersDic.ContainsKey(currentKey))
-                    {
-                        headersDic.Add(currentKey, value);
-                    }
-                    else
+                    currentKey = currentKey.ToLowerInvariant();
+                    var value = getHeaderValue(currentKey);
+
+                    if (value is not null)
                     {
-                        Log.Warning("Header {Key} couldn't be added as argument to the waf", currentKey);
+                        if (!headersDic.ContainsKey(currentKey))
```

Comment:
You should be able to use `headersDic.TryAdd()` here instead of the two separate calls to `ContainsKey()` and `Add()`.

Reply:
Oops, I forgot that `TryAdd()` was added in .NET Core. I was thinking of `TryGetValue()`. Feel free to ignore, then!